pytorch_lightning/trainer/trainer.py

```diff
@@ -536,6 +536,8 @@ def fit(
 
         self._run(model)
 
+        self.data_connector.detach_data(model)
```

Comment:
Why not put these calls inside `_run` if all call it?

Reply:
I thought about that, but since attaching also happens outside of run, I chose to have it on the same level. can also move it though

Reply:
i think it makes sense to include in `_run` so it's shared. specifically, does this belong in the teardown hook?

Reply:
moved to teardown